### PR TITLE
Refactor Bridge related tests

### PR DESCRIPTION
**Pull request by marcos-iov**
- Remove unthrown exceptions from the methods declaration
- Rename variables
- Move utils methods to it's own class

### CODE REVIEW

**Review by jeremy-then — APPROVED** [review]
LGTM.

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java [hunk 1]
```diff
@@ -106,34 +115,48 @@ class BridgeSupportTest {
...
         signatureCache = new BlockTxSignatureCache(new ReceivedTxSignatureCache());
+
         bridgeSupportBuilder = new BridgeSupportBuilder();
-        StorageAccessor inMemoryStorageAccessor = new InMemoryStorage();
-        whitelistStorageProvider = new WhitelistStorageProviderImpl(inMemoryStorageAccessor);
+
+        StorageAccessor bridgeStorageAccessor = new InMemoryStorage();
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(bridgeStorageAccessor);
+        whitelistStorageProvider = new WhitelistStorageProviderImpl(bridgeStorageAccessor);
+        LockingCapStorageProvider lockingCapStorageProvider = new LockingCapStorageProviderImpl(bridgeStorageAccessor);
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java:134]: isn't kind of weird to have the whitelistStorageProvider defined as class level but the other ones not?

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 2]
```diff
@@ -458,7 +446,7 @@ void getNewFederationBtcUTXOs(
 
     @Test
     void getNewFederationBtcUTXOs_calledTwice_returnCachedUtxos() {
-        ActivationConfig.ForBlock activations = ActivationConfigsForTest.only(ConsensusRule.RSKIP123).forBlock(0);
+        ActivationConfig.ForBlock wasabiActivations = ActivationConfigsForTest.wasabi100().forBlock(0);
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:449]: why wasabi? we are in mainnet so the storage key is always [this one](https://github.com/rsksmart/rskj/pull/2706/files#diff-f42b277f39730e846e15f899b88e1e7e718280d3ea3d4c4d5b035a5d66385923R455) right?

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java [hunk 3]
```diff
@@ -17,179 +17,156 @@
...
 
     private List<LogInfo> eventLogs;
     private BridgeEventLogger eventLogger;
-    private BtcTransaction btcTxMock;
-    private BtcTransaction btcTx;
     private SignatureCache signatureCache;
 
     @BeforeEach
     void setup() {
+        ActivationConfig.ForBlock activations = ActivationConfigsForTest.lovell700().forBlock(0L);
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java:74]: should we use `all` instead of `lovell`?

**Review by apancorb — APPROVED** [review]
Nice job! LGTM

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java (hunk 3 above)
Comment (marcos-iov) [reply rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java]: True, updated

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java [hunk 4]
```diff
@@ -225,53 +202,60 @@ private static Stream<Arguments> logAddSignatureArgProvider() {
...
+            new Object[]{BTC_TRANSACTION.bitcoinSerialize()}
+        );
     }
 
     @ParameterizedTest
     @ValueSource(booleans = {false, true})
     void logCommitFederation(boolean isRSKIP383Active) {
-        when(activations.isActive(ConsensusRule.RSKIP383)).thenReturn(isRSKIP383Active);
+        ActivationConfig.ForBlock hopActivations = ActivationConfigsForTest.hop400().forBlock(0);
+        ActivationConfig.ForBlock lovellActivations = ActivationConfigsForTest.lovell700().forBlock(0L);
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java:249]: shouldn't we use use all activations here?

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 5]
```diff
@@ -293,8 +280,7 @@ private static Stream<Arguments> provideSaveFederationTestArguments() {
     @Test
     void saveNewFederation_before_RSKIP123_should_not_save_null() {
         Repository repository = mock(Repository.class);
-        activations = mock(ActivationConfig.ForBlock.class);
-        when(activations.isActive(ConsensusRule.RSKIP123)).thenReturn(false);
+        activations = ActivationConfigsForTest.orchid().forBlock(0L);
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:283]:
```suggestion
        ActivationConfig.ForBlock orchidActivations = ActivationConfigsForTest.orchid().forBlock(0L);
```
wdyt? To be consistent with other tests

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 6]
```diff
@@ -941,7 +926,7 @@ private static Federation createNonStandardErpFederation() {
             federationArgs,
             erpPubKeys,
             activationDelay,
-            activations
+            ActivationConfigsForTest.all().forBlock(0L)
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:929]: use class level activations?

File: rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java (hunk 4 above)
Comment (marcos-iov) [reply rskj-core/src/test/java/co/rsk/peg/utils/BridgeEventLoggerImplTest.java]: Done

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 6 above)
Comment (marcos-iov) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:929]: Class level activations gets modified in a couple of places, creates some issues here

Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:929]: oki doki

**Review by josedahlquist — APPROVED** [review]

**Pull request merged**